### ISSUE TITLE
The env list obtained through velaux is out of sync with the env list obtained through vela cli

### ISSUE THREAD

**Issue by nuclearwu** [kubevela/kubevela#4776]
**Describe the bug**
The environment list obtained through velaux is out of sync with the environment list obtained through vela cli. I created an env through vela cli, but I can't see it in velaux's environment list. Instead, I created the env via velaux, which is visible via the vela cli，but when I delete it via vela cli, it is still in velaux's env list

![image](https://user-images.githubusercontent.com/37976659/191912172-784e957e-bcfe-41d4-bece-8a4c1950f19f.png)
![image](https://user-images.githubusercontent.com/37976659/191912719-d9e3cb63-0c7e-4099-8939-1779ffca11a0.png)

![image](https://user-images.githubusercontent.com/37976659/191914653-d6049d68-3549-4976-96a2-fc54349129be.png)
![image](https://user-images.githubusercontent.com/37976659/191914704-28d040dd-1f98-4066-b3d2-1eba1ff252f9.png)

**To Reproduce**

**Expected behavior**
The env list obtained through velaux should be in sync with the env list obtained through vela cli

**Screenshots**

**KubeVela Version**
1.5.4

**Cluster information**
Kubernetes 1.21.5

**Additional context**

**Comment by nuclearwu** [comment]
I checked the source code, and the env list obtained by velaux actually queries the configmap under the kubevela namespace, while the environment list obtained by vela cli actually queries the namespace with the env label. Is this a bug or is it designed that way?

**Comment by wonderflow** [comment]
@nuclearwu Thanks for pointing that, I think we need to fix it, both velaux and vela CLI should align with the K8s namespace and label mechanism without using extra Configmap.  \cc @barnettZQG

**Comment by nuclearwu** [comment]
> @nuclearwu Thanks for pointing that, I think we need to fix it, both velaux and vela CLI should align with the K8s namespace and label mechanism without using extra Configmap. \cc @barnettZQG

If this is a problem, I'm happy to fix it

**Comment by nuclearwu** [comment]
I also have a problem that velaux has the concept of delivery target (actually created configmap), and velaux's env is bound to delivery target, but vela cli does not, so if both velaux and vela cli are are align with the k8s namespace and label mechanism, what about the delivery target? Delivery target related commands not found in vela cli.

**Comment by wonderflow** [comment]
>  what about the delivery target? Delivery target related commands not found in vela cli.

Yes, we don't have delivery target in CLI, it's a special concept of velaux

**Comment by nuclearwu** [comment]
> 

VelaCLI initializes the env's NAME and NAMESPACE when the env is created, while VelaUX initializes the Name, Project, Namespace, Description and Targets when the env is created. So in the process of synchronizing the env of velaux and velacli, are all these information(such as Project, Description and Targets) bound in the annotation of NAMESPACE?

**Comment by nuclearwu** [comment]
> > what about the delivery target? Delivery target related commands not found in vela cli.
> 
> Yes, we don't have delivery target in CLI, it's a special concept of velaux. You can mark the bound information inside the annotation of namespace.

VelaCLI initializes the env's NAME and NAMESPACE when the env is created, while VelaUX initializes the Name, Project, Namespace, Description and Targets when the env is created. So in the process of synchronizing the env of velaux and velacli, are all these information(such as Project, Description and Targets) bound in the annotation of NAMESPACE?

**Comment by wonderflow** [comment]
> VelaCLI initializes the env's NAME and NAMESPACE when the env is created, while VelaUX initializes the Name, Project, Namespace, Description and Targets when the env is created. So in the process of synchronizing the env of velaux and velacli, are all these information(such as Project, Description and Targets) bound in the annotation of NAMESPACE?

I think that could be fine. WDYT? \cc @barnettZQG

**Comment by barnettZQG** [comment]
@nuclearwu We could get all env namespaces in listing the environments API, and create a target base on the same namespace default.

**Comment by nuclearwu** [comment]
> > VelaCLI initializes the env's NAME and NAMESPACE when the env is created, while VelaUX initializes the Name, Project, Namespace, Description and Targets when the env is created. So in the process of synchronizing the env of velaux and velacli, are all these information(such as Project, Description and Targets) bound in the annotation of NAMESPACE?
> 
> I think that could be fine. WDYT? \cc @barnettZQG

I agree with you, but at present, the code of ENVService involves methods such as GetEnv, ListEnvs, ListEnvCount, DeleteEnv, CreateEnv, UpdateEnv, etc. that need to be rewritten, more like a feature needs to be submitted.  @wonderflow @barnettZQG

**Comment by briankane** [comment]
This issue has been inactive for over 2 years and is scheduled for closure as part of our backlog maintenance.

  **If this issue is still relevant:**
  - Please comment below to keep it open
  - Provide updated context if the original description is outdated

  **Timeline:** If there's no activity by **Friday, October 24th**, this issue will be automatically closed.

  **Note:** Issues can always be reopened if they become relevant again. We encourage the community to reopen and reprioritize any incorrectly 
  closed issues.

**Comment by briankane** [comment]
This issue has been closed following the stale issue notice. 

If this issue is still relevant and should remain open, please:
1. Reopen the issue
2. Remove the 'stale' label
3. Provide an update on the current status or importance

Thank you for your understanding and contribution to KubeVela!
